includes/abstracts/abstract-wc-order.php

```diff
@@ -122,7 +122,7 @@ public function __construct( $order = 0 ) {
 		if ( is_numeric( $order ) && $order > 0 ) {
 			$this->set_id( $order );
 		} elseif ( $order instanceof self ) {
-			$this->set_id( absint( $order->get_id() ) );
+			$this->set_id( $order->get_id() );
 		} elseif ( ! empty( $order->ID ) ) {
 			$this->set_id( absint( $order->ID ) );
```

Comment:
Yeah, thats not trusted since it's a property from another 'thing'. It's not an WC Order object.